src/validator-registry-tp/src/validator_registry_tp.rs

```diff
@@ -166,55 +167,116 @@ impl ValidatorRegistryTransactionHandler {
-	self._set_data(context, &validator_map_address, 
-			&serde_json::to_string(&validator_map).unwrap());
+        let validator_map_address = _get_address(&String::from("validator_map"));
+        self._set_data(context, &validator_map_address,
+                       &serde_json::to_string(&validator_map).unwrap());
 
         info!("Validator id {} was added to the validator_map and set.",
-                validator_id);
+              validator_id);
 
-         Ok(())
+        Ok(())
     }
 
-    fn _set_data( &self, context : &mut TransactionContext,
-                         address : &String,
-                            data : &String, ) -> () {
-
-        let mut map:HashMap<String, Vec<u8>> = HashMap::new();
+    fn _set_data(&self, context: &mut TransactionContext,
+                 address: &String,
+                 data: &String, ) -> () {
+        let mut map: HashMap<String, Vec<u8>> = HashMap::new();
         map.insert(address.to_string(), data.as_bytes().to_vec());
         let addresses = context.set_state(map);
-        if addresses.is_err(){
+        if addresses.is_err() {
             warn!("Failed to save value at address {}", address);
         }
     }
 
     fn _verify_signup_info(&self,
-                    originator_public_key_hash: &String,
-                    val_reg_payload: &ValidatorRegistryPayload,
-                    context: &mut TransactionContext,) -> Result <(), ValueError> {
-         let signup_info : ValidatorRegistrySignupInfo = serde_json::from_str(val_reg_payload.signup_info_str).unwrap();
-         let proof_data : SignupInfoProofData = serde_json::from_str(&*signup_info.proof_data).unwrap();
-
-         let report_public_key_pem = self._get_config_setting(context, "sawtooth.poet.report_public_key_pem").unwrap();
-         
-         Ok(())
+                           originator_public_key_hash: &String,
+                           val_reg_payload: &ValidatorRegistryPayload,
+                           context: &mut TransactionContext, ) -> Result<(), ValueError> {
+        let signup_info: ValidatorRegistrySignupInfo = serde_json::from_str(&val_reg_payload
+            .signup_info_str).unwrap();
+        let proof_data: SignupInfoProofData = serde_json::from_str(&*signup_info.proof_data).unwrap();
+        // Verify the attestation verification report signature
+        let verification_report = proof_data.verification_report;
+        let signature = proof_data.signature;
+
+        // Try to get the report key from the configuration setting.  If it
+        // is not there or we cannot parse it, fail verification.
+        let report_public_key_pem = self._get_config_setting(context, &"sawtooth.poet\
+         .report_public_key_pem".to_string()).unwrap();
+        let valid_measurements = self._get_config_setting(context, &"sawtooth.poet\
+         .valid_enclave_measurements".to_string()).unwrap();
+//        let valid_enclave_measurements = [bytes.fromhex(m)
```

Comment:
Why is this line commented?

Reply:
It's used for quote verification, the part which is pending.